src/transformers/pipelines/text_generation.py

```diff
@@ -107,6 +109,13 @@ def _sanitize_parameters(
                     " [None, 'hole']"
                 )
             preprocess_params["handle_long_generation"] = handle_long_generation
+        if stop_sequence is not None:
+            stop_sequence_ids = self.tokenizer.encode(
+                stop_sequence
+            )
+            if len(stop_sequence_ids) > 1:
+                warnings.warn(f"Stopping on a multiple token sequence is not yet supported on transformers. The first token of the stop sequence will be used as the stop sequence string in the interim.")
+            generate_kwargs["eos_token_id"] = stop_sequence_ids[0]
```

Comment:
I think implementing either in `generate` or here is enough.
We shouldn't try to implement it everywhere.

@gante @patrickvonplaten Are you ok if it's directly included in `generate` (otherwise we can keep it just in the pipeline).